### ISSUE TITLE
azurerm_app_configuration not returning app configurations / referenced key vault secrets created using terraform.

### ISSUE THREAD

**Issue by White7292** [hashicorp/terraform-provider-azurerm#19242]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.3.2 on windows_amd64

### AzureRM Provider Version

3.30.0

### Affected Resource(s)/Data Source(s)

azurerm_app_configuration azurerm_app_configuration_key azurerm_key_vault azurerm_key_vault_secret

### Terraform Configuration Files

```hcl
# Terraform Configuration
# =========================
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "3.30.0"
    }
  }
}

# Providers
# =========================
provider "azurerm" {
  features {}
}

# Client Configuration
# =========================
data "azurerm_client_config" "current" {}


# Resource Group
# =========================
resource "azurerm_resource_group" "rg" {
  name     = "rg-test-26245672546"
  location = "westus2"
}


# Role Assignments
# =========================
resource "azurerm_role_assignment" "app_configuration_data_owner" {
  scope                = azurerm_app_configuration.app_configuration.id
  role_definition_name = "App Configuration Data Owner"
  principal_id         = data.azurerm_client_config.current.object_id
}


# Key Vault
# =========================
resource "azurerm_key_vault" "key_vault" {
  name                       = "kv-test-26245672546"
  location                   = azurerm_resource_group.rg.location
  resource_group_name        = azurerm_resource_group.rg.name
  tenant_id                  = data.azurerm_client_config.current.tenant_id
  sku_name                   = "standard"
  soft_delete_retention_days = 7

  access_policy {
    tenant_id = data.azurerm_client_config.current.tenant_id
    object_id = data.azurerm_client_config.current.object_id

    key_permissions = [
      "Create",
      "Get",
    ]

    secret_permissions = [
      "List",
      "Set",
      "Get",
      "Delete",
      "Purge",
      "Recover"
    ]
  }

  tags = {
    Environment     = "DEV"
    ApplicationName = "TEST"
  }
}


# App Configuration
# =========================
resource "azurerm_app_configuration" "app_configuration" {
  name                = "config-test-26245672546"
  resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location

  tags = {
    Environment     = "DEV"
    ApplicationName = "TEST"
  }
}


# Storage
# =========================
resource "azurerm_storage_account" "storage_account" {
  name                     = "sttest26245672546"
  resource_group_name      = azurerm_resource_group.rg.name
  location                 = azurerm_resource_group.rg.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
  access_tier              = "Cool"

  tags = {
    Environment     = "DEV"
    ApplicationName = "TEST"
  }
}

data "azurerm_storage_account" "storage_account" {
  name                = azurerm_storage_account.storage_account.name
  resource_group_name = azurerm_storage_account.storage_account.resource_group_name
}


# App Configuration / Secrets
# =========================
# Test App Configuration.
resource "azurerm_app_configuration_key" "app_config_test" {
  configuration_store_id = azurerm_app_configuration.app_configuration.id
  key                    = "TestKey"
  type                   = "kv"
  label                  = "Test KV from Terraform."
  value                  = "This comes from Terraform."

  depends_on = [
    azurerm_role_assignment.app_configuration_data_owner,
  ]
}

# Test App Configuration that references Key Vault secret.
resource "azurerm_key_vault_secret" "kv_secret_storage_account_connection_string" {
  key_vault_id = azurerm_key_vault.key_vault.id

  name         = "AzureBlobStorageConfigs--ConnectionString"
  value        = data.azurerm_storage_account.storage_account.primary_connection_string
}

resource "azurerm_app_configuration_key" "app_config_kv_reference_storage_account_connection_string" {
  configuration_store_id = azurerm_app_configuration.app_configuration.id
  key                    = "AzureBlobStorageConfigs:ConnectionString"
  type                   = "vault"
  label                  = "Connection string for ${"DEV"} Azure Storage Account."
  vault_key_reference    = azurerm_key_vault_secret.kv_secret_storage_account_connection_string.id

  depends_on = [
    azurerm_role_assignment.app_configuration_data_owner
  ]
}
```

### Debug Output/Panic Output

```shell
Apply complete! Resources: 8 added, 0 changed, 0 destroyed.
```

### Expected Behaviour

Seeing as I effectively copy and pasted the [Example Usage]'s (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_secret#example-usage) for these Terraform Resources, I'm expecting that the code (`AzureAppConfigurationProvider`) should be able to fetch the `AzureBlobStorageConfigs:ConnectionString` as well as the `TestKey` values from the Terraform provisioned Azure App Configuration.

### Actual Behaviour

The `AzureBlobStorageConfigs:ConnectionString` & `TestKey` values are not returned via the `AzureAppConfigurationProvider` by the Terraformed provisioned Azure App Configuration. But when I manually create a key value pair as well as a new Key Vault Reference (that references the Terraform-created Key Vault Secret), the manually created values are returned (and still no terraform-created App Configuration values).

### Steps to Reproduce

`terraform apply -auto-approve`

Directly after provisioning managed infrastructure using Terraform...
Secret was created in key vault:
![Pasted image 20221110143700](https://user-images.githubusercontent.com/16881776/201199079-1ee4e4af-ccb4-4594-b92f-9fb7a2987797.png)

App configuration references key vault secret as well as one app config:
![Pasted image 20221110143836](https://user-images.githubusercontent.com/16881776/201199109-94f510ea-abc4-4ac8-abe5-d8f25ab51c5a.png)

No values are returned by the `AzureAppConfigurationProvider` in the code:
![Pasted image 20221110143607](https://user-images.githubusercontent.com/16881776/201199154-a4b83ffd-de24-4751-9e76-5c82359a8a6b.png)

I was confused as to why I wasn't receiving any values... It could be that I wasn't connecting to the Azure App Configuration, or that there was an issue with the Key Vault Reference that was silently throwing an exception. So I went ahead and manually created a key value pair and I referenced the exact same Azure Key Vault Secret to see if I could get those values.

Manually creating new App Configuration Key Vault Reference:
![Pasted image 20221110144028](https://user-images.githubusercontent.com/16881776/201199221-527935d9-7e27-4393-8f87-39ffa50e4ec4.png)

Manually creating new App Configuration key value pair:
![Pasted image 20221110144128](https://user-images.githubusercontent.com/16881776/201199243-9e5eb0b2-f5c0-4c4a-89a2-ba8626b67b8c.png)

New state of App Configuration (showing both the Terraform created app configurations and the manually created ones):
![Pasted image 20221110144149](https://user-images.githubusercontent.com/16881776/201199257-2cb57429-b3b1-4e5e-8345-65f697621ebb.png)

Only manually created values are returned by the `AzureAppConfigurationProvider` in the code:
![Pasted image 20221110144508](https://user-images.githubusercontent.com/16881776/201199277-ce841c01-8e96-424b-ad63-0371083cb203.png)

So the code can fetch some of the App Configurations- but not the ones created during the `Terraform apply` phase.

**Comment by teowa** [comment]
Hi @White7292 , thanks for submitting this issue!
I have tried to reproduce the issue, I can successfully read all 4 key-value pairs with [`azurerm_app_configuration_keys`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/app_configuration_keys) data source. 
Since I am not familiar with the metioned `AzureAppConfigurationProvider` code, from the behaviour only part of the values are returned, could it be some filters that omit part of the results? (I also notice that the Manually created key values have no `label`)

**Comment by White7292** [comment]
Hey @teowa, thanks for the quick reply!

The `[AzureAppConfigurationProvider](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.AzureAppConfiguration/)` (written by Microsoft) is the "configuration provider for the .NET Core framework that allows developers to use Microsoft Azure App Configuration service as a configuration source in their applications."

There seems to be a difference between the secrets/app configuration key-value pairs created by Terraform and created my myself- at least in the eyes of Microsoft's own code.

There's no filters on the values I'm trying to fetch- the code is set to fetch everything:
```
configurationManager.AddAzureAppConfiguration(options => {
    options.Connect(
            new Uri(azureAppConfigurationUrl),
            credentials
        )
        .ConfigureKeyVault(options => {
            options.SetCredential(credentials);
        });
});
```

I'm quite puzzled though, because when I manually create a new Key Vault Reference, the `AzureAppConfigurationProvider` is able to fetch the Terraform-created [azurerm_key_vault_secret](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_secret), where as the Terraform-created [azurerm_app_configuration_key](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_configuration_key) isn't even present...

All of which leads me to believe that the way `azurerm_app_configuration_key` creates an App Configuration isn't the same as creating one via the UI? 

I could definitely be wrong here-

**Comment by CorrenSoft** [comment]
@teowa hit the point.
The miss interpretation here is that the label field is not intended as an "informative description" but as a per-environment classification (that means, you can have the same key multiple times with different labels). And by default, the SDK is excluding the key-value pairs that are having a label.
For more info about it, you can check [here](https://learn.microsoft.com/en-us/azure/azure-app-configuration/howto-labels-aspnet-core?tabs=core5x#load-configuration-values-with-a-specified-label).

In short, remove the label and you will be able to read the values (please, confirm).

**Comment by White7292** [comment]
Wow! I had assumed that `Label` was a description of the app configuration- thanks for clearing that up!

I removed the labels from my [`azurerm_app_configuration_key`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_configuration_key) resources, and the problem is solved!

![image](https://user-images.githubusercontent.com/16881776/202002108-a604f5f7-9f6c-47c5-8e30-f350b9cb14e3.png)

Thank you both @teowa & @CorrenSoft for the help (as well as quick replies)!
